src/Model.ts

```diff
@@ -553,10 +553,11 @@ export default class Model<S extends SyncFuncConstraint> extends EventEmitter<Re
 
     const modelsEvent: ConfirmedEvent = {
       ...event!,
+      name: event?.name || '',
       confirmed: true,
       rejected,
       mutationId: mutationId,
-      sequenceId: event.id,
+      sequenceId: event?.id || '',
```

Comment:
Do we want to fail if the sequenceId is null / undefined or coerce the value to '0'?

Reply:
Here we're dealing with sequenceIds that have come from the channel messages. The event will always have an ID,because ably will assign it a random one if the customer doesn't assign the ID on publish.

Reply:
Good point. I wonder why the SDK v2 has this an optional param. I suppose everything is option in JS. I'll leave the current check in then